src/openrct2/core/Imaging.cpp

```diff
@@ -199,9 +199,9 @@ namespace Imaging
                 {
                     throw std::runtime_error("png_malloc failed.");
                 }
-                for (size_t i = 0; i < PNG_MAX_PALETTE_LENGTH; i++)
+                for (uint16_t i = 0; i < PNG_MAX_PALETTE_LENGTH; i++)
                 {
-                    const auto& entry = (*image.Palette)[static_cast<uint16_t>(i)];
+                    const auto& entry = (*image.Palette)[i];
```

Comment:
This seems like a random change, there is also no need to cast the index to uint16_t, not sure why that was done to begin with.

Reply:
There apparently _is_ a need to do that - with the cast removed it now produces compiler warnings.

Reply:
This is pretty weird, I guess there is some operator overloading going on here which doesn't adhere the standards.

Reply:
Seems a little bit outside of the remit of this PR, then.

@Harry-Hopkinson Could you revert this line altogether (i.e. restore the `static_cast`)?

Reply:
@Harry-Hopkinson I see you applied most of Matt’s requests, but you still need to update the `PALETTE_SIZE` constant. Could you do that?

Reply:
I initially changed it but it was creating lots of errors, I think I know what it up.